api/src/pcapi/core/offers/models.py

```diff
@@ -574,13 +574,17 @@ def isFromAllocine(self) -> bool:
     def isFromCinemaProvider(self) -> bool:
         return self.isFromProvider and self.lastProvider.isCinemaProvider
 
-    @property
+    @hybrid_property
     def isBookable(self) -> bool:
         for stock in self.stocks:
             if stock.isBookable:
                 return True
         return False
 
+    @isBookable.expression  # type: ignore [no-redef]
+    def isBookable(cls) -> bool:  # pylint: disable=no-self-argument
+        return sa.exists().where(Stock.offerId == cls.id).where(Stock._bookable)
```

Comment:
Ça ne me semble tout à fait équivalent à la propriété (Python) `isBookable()`. Celle-ci utilise `Stock.isBookable()` qui renvoie vraie si `stock._bookable and offer.isReleased`. Pour la propriété "SQL", il manque la seconde partie de la condition (`isReleased`).

Reply:
Finalement j'ai utilisé is_eligible_for_search juste en dessous, elle fait ce que je veux